src/python/pants/option/arg_splitter_test.py

```diff
@@ -297,10 +306,10 @@ def help_no_arguments_test(command_line: str, **expected):
         help_no_arguments_test("./pants -h"),
         help_no_arguments_test("./pants --help"),
         help_no_arguments_test("./pants help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants help --help-advanced", expected_help_advanced=True),
         help_no_arguments_test("./pants --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help --help-advanced", expected_help_advanced=True),
-        help_no_arguments_test("./pants --help-advanced --help", expected_help_advanced=True),
```

Comment:
I find these "help --help-advanced" variants a bit flimsy. But if it really is desirable to support that, we'll have to look at the `builtin_goal_args` in the help builtin class to support that style now. (as we break out of goal parsing in the arg splitter on the first occurence of a built in goal).

Reply:
Yeah I don't really know why this use case matters. I think it's reasonable to go with first encountered.